src/v/redpanda/admin/server.cc

```diff
@@ -1860,6 +1863,21 @@ admin_server::patch_cluster_config_handler(
                       yaml_name,
                       property.format_raw(yaml_value),
                       validation_err.value().error_message());
+                } else if (auto restricted_err = property.check_restricted(val);
+                           restricted_err.has_value() && should_sanction) {
```

Comment:
to understand the scope of the configuration system changes, i think i'm interested in the following: does this do something fundamentally different than what could be accomplished if the interface where something like

```
bool check_restricted(feature_table, current_config, proposed_update);
```

where this function check_restricted had all the enterprise policies to accept/reject a proposed update given the current system state, as opposed to annotating the individual properties.

Reply:
>does this do something fundamentally different

No not really. All this does is make each property instance self-describing with respect to which values are "special".